# awesome-protein-representation-learning

![PRs Welcome](https://img.shields.io/badge/PRs-Welcome-green)  [![Awesome](https://awesome.re/badge.svg)](https://awesome.re) ![Stars](https://img.shields.io/github/stars/LirongWu/awesome-protein-representation-learning?color=yellow)  ![Forks](https://img.shields.io/github/forks/LirongWu/awesome-protein-representation-learning?color=blue&label=Fork)

This repository contains a list of papers on the **Protein Representation Learning (PRL)**, we categorize them based on their published years. We will try to make this list updated. If you found any error or any missed paper, please don't hesitate to open issues or pull requests.

## Year 2023
1. [NIPS 2023] **Injecting Multimodal Information into Rigid Protein Docking via Bi-level Optimization**[[paper]](https://openreview.net/forum?id=ZuaVKlWdD2)
1. [NIPS 203] **Unsupervised Protein-Ligand Binding Energy Prediction via Neural Euler's Rotation Equation**[[paper]](https://arxiv.org/abs/2301.10814)[[code]](github.com/wengong-jin/DSMBind)
1. [NIPS 2023] **ProteinInvBench: Benchmarking Protein Inverse Folding on Diverse Tasks, Models, and Metrics**[[paper]](https://nips.cc/virtual/2023/poster/73537)[[code]](https://github.com/A4Bio/ProteinInvBench)
1. [NIPS 2023] **OpenProteinSet: Training data for structural biology at scale**[[paper]](https://arxiv.org/abs/2308.05326)
1. [NIPS 2023] **CELLE-2: Translating Proteins to Pictures and Back with a Bidirectional Text-to-Image Transformer**[[paper]](https://paperswithcode.com/paper/celle-2-translating-proteins-to-pictures-and)[[code]](https://github.com/BoHuangLab/CELL-E_2)
1. [NIPS 2023] **FABind: Fast and Accurate Protein-Ligand Binding**[[paper]](https://arxiv.org/abs/2310.06763)[[code]](https://github.com/qizhipei/fabind)
1. [NIPS 2023] **ProteinGym: Large-Scale Benchmarks for Protein Design and Fitness Prediction**[[paper]](https://www.biorxiv.org/content/10.1101/2023.12.07.570727v1)[[code]](https://github.com/OATML-Markslab/ProteinGym)
1. [NIPS 2023] **Graph Denoising Diffusion for Inverse Protein Folding**[[paper]](https://arxiv.org/abs/2306.16819)[[code]](https://github.com/ykiiiiii/GraDe_IF)
1. [NIPS 2023] **DiffPack: A Torsional Diffusion Model for Autoregressive Protein Side-Chain Packing**[[paper]](https://arxiv.org/abs/2306.01794)[[code]](https://github.com/DeepGraphLearning/DiffPack)
1. [NIPS 2023] **DrugCLIP: Contrasive Protein-Molecule Representation Learning for Virtual Screening**[[paper]](https://www.microsoft.com/en-us/research/publication/drugclip-contrasive-protein-molecule-representation-learning-for-virtual-screening/)
1. [NIPS 2023] **Predicting mutational effects on protein-protein binding via a side-chain diffusion probabilistic model**[[paper]](https://synthical.com/article/8e405243-504c-40a5-907e-c9d1539bb0bc)
1. [NIPS 2023] **PoET: A generative model of protein families as sequences-of-sequences**[[paper]](https://arxiv.org/abs/2306.06156)[[code]](https://github.com/OpenProteinAI/PoET)
1. [NIPS 2023] **Pre-Training Protein Encoder via Siamese Sequence-Structure Diffusion Trajectory Prediction**[[paper]](https://openreview.net/forum?id=tIzbNQko3c)
1. [ICML 2023] **Bidirectional Learning for Offline Model-based Biological Sequence Design**[[paper]](https://openreview.net/forum?id=CUORPu6abU)[[code]](https://github.com/GGchen1997/BIB-ICML2023-Submission)
1. [ICML 2023] **Reprogramming Pretrained Language Models for Antibody Sequence Infilling**[[paper]](https://openreview.net/forum?id=K2gn1WiLAu)[[code]](https://github.com/IBM/ReprogBERT)
1. [ICML 2023] **SE(3) diffusion model with application to protein backbone generation**[[paper]](https://openreview.net/forum?id=m8OUBymxwv)[[code]](https://github.com/jasonkyuyim/se3_diffusion)
1. [ICML 2023] **Importance Weighted Expectation-Maximization for Protein Sequence Design**[[paper]](https://openreview.net/forum?id=7viOT7Zs9G)[[code]](https://github.com/JocelynSong/IsEM-Pro.git)
1. [ICML 2023] **Generating Novel, Designable, and Diverse Protein Structures by Equivariantly Diffusing Oriented Residue Clouds**[[paper]](https://openreview.net/forum?id=4Kw5hKY8u8)[[code]](https://github.com/aqlaboratory/genie)
1. [ICML 2023] **Chemically Transferable Generative Backmapping of Coarse-Grained Proteins**[[paper]](https://openreview.net/forum?id=7DnvWyVkUo)[[code]](https://github.com/learningmatter-mit/GenZProt)
1. [ICML 2023] **Structure-informed Language Models Are Protein Designers**[[paper]](https://openreview.net/forum?id=1F2Opw8CGA)
1. [ICML 2023] **ProtST: Multi-Modality Learning of Protein Sequences and Biomedical Texts** [[paper]](https://arxiv.org/abs/2301.12040)[[code]](https://github.com/DeepGraphLearning/ProtST)
1. [bioRxiv 2023] **Retrieved Sequence Augmentation for Protein Representation Learning** [[paper]](https://www.biorxiv.org/content/10.1101/2023.02.22.529597v2)[[code]](https://github.com/HKUNLP/RSA) 
1. [Arxiv 2023] **Data-Efficient Protein 3D Geometric Pretraining via Refinement of Diffused Protein Structure Decoy** [[paper]](https://arxiv.org/pdf/2302.10888.pdf)
1. [ICLR 2023] **Protein Representation Learning by Geometric Structure Pretraining** [[paper]](https://arxiv.org/abs/2203.06125)
3. [ICLR 2023] **Protein Representation Learning via Knowledge Enhanced Primary Structure Reasoning**[[paper]](https://openreview.net/forum?id=VbCMhg7MRmj)
4. [ICLR 2023] **Multi-level Protein Structure Pre-training via Prompt Learning**[[paper]](https://openreview.net/forum?id=XGagtiJ8XC)
5. [ICLR 2023] **Learning Hierarchical Protein Representations via Complete 3D Graph Networks**[[paper]](https://openreview.net/forum?id=9X-hgLDLYkQ)
5. [ICLR 2023] **Rotamer Density Estimator is an Unsupervised Learner of the Effect of Mutations on Protein-Protein Interaction**[[paper]](https://openreview.net/forum?id=_X9Yl1K2mD)[[code]](https://github.com/luost26/RDE-PPI)
5. [ICLR 2023] **Matching receptor to odorant with protein language and graph neural networks**[[paper]](https://openreview.net/forum?id=q9VherQJd8_)[[code]](https://github.com/MatejHl/Receptor2Odorant)
5. [ICLR 2023] **Protein Sequence and Structure Co-Design with Equivariant Translation**[[paper]](https://openreview.net/forum?id=pRCMXcfdihq)[[code]](https://github.com/sokrypton/ColabDesign)
5. [ICLR 2023] **HotProtein: A Novel Framework for Protein Thermostability Prediction and Editing**[[paper]](https://openreview.net/pdf?id=YDJRFWBMNby)[[code]](https://github.com/VITA-Group/HotProtein)
5. [ICLR 2023] **PiFold: Toward effective and efficient protein inverse folding**[[paper]](https://openreview.net/forum?id=oMsN9TYwJ0j)[[code]](https://github.com/A4Bio/PiFold)
5. [ICLR 2023] **Continuous-Discrete Convolution for Geometry-Sequence Modeling in Proteins**[[paper]](https://openreview.net/forum?id=P5Z-Zl9XJ7)[[code]](https://github.com/hehefan/Continuous-Discrete-Convolution)
5. [Nature 2023] **De novo design of protein interactions with learned surface fingerprints**[[paper]](https://www.nature.com/articles/s41586-023-05993-x)[[code]](https://github.com/LPDI-EPFL/masif_seed)
5. [Nature Communications 2023] **PeSTo: parameter-free geometric deep learning for accurate prediction of protein binding interfaces** [[paper]](https://www.nature.com/articles/s41467-023-37701-8)[[code]](https://github.com/LBM-EPFL/PeSTo)
16. [IJCAI 2023] **SemiGNN-PPI: Self-Ensembling Multi-Graph Neural Network for Efficient and Generalizable Protein-Protein Interaction Prediction** [[paper]](https://arxiv.org/abs/2305.08316)
17. [Science 2023] **Top-down design of protein architectures with reinforcement learning**[[paper]](https://www.science.org/doi/abs/10.1126/science.adf6591)[[code]](https://github.com/idlutz/protein-backbone-MCTS)
18. [Nature Communications 2023] **Hierarchical graph learning for protein–protein interaction** [[paper]](https://www.nature.com/articles/s41467-023-36736-1)[[code]](https://github.com/zqgao22/HIGH-PPI)
18. [Nature Biotechnol 2023] **Efficient evolution of human antibodies from general protein language models** [[paper]](https://www.nature.com/articles/s41587-023-01763-2)[[code]](https://github.com/brianhie/efficient-evolution)
18. [Advanced Science 2023] **A Multimodal Protein Representation Framework for Quantifying Transferability Across Biochemical Downstream Tasks** [[paper]](https://onlinelibrary.wiley.com/doi/full/10.1002/advs.202301223)[[code]](https://github.com/SIAT-code/MASSA)
19. [Nature Communications Biology] **Learning the protein language of proteome-wide protein-protein binding sites via explainable ensemble deep learning**[[paper]](https://doi.org/10.1038/s42003-023-04462-5)[[code]](https://github.com/houzl3416/EDLMPPI.git)

## Year 2022
1. [Nature Communications 2022]**ProtGPT2 is a deep unsupervised language model for protein design**[[paper]](https://www.nature.com/articles/s41467-022-32007-7)
1. [bioRxiv 2022] **Codon language embeddings provide strong signals for protein engineering** [[paper]](https://www.biorxiv.org/content/10.1101/2022.12.15.519894v1)
1. [Arxiv 2022] **When Geometric Deep Learning Meets Pretrained Protein Language Models** [[paper]](https://arxiv.org/abs/2212.03447)
1. [Arxiv 2022] **Contrastive Representation Learning for 3D Protein Structures** [[paper]](https://arxiv.org/abs/2205.15675)
1. [Bioinformatics 2022] **Structure-aware Protein Self-supervised Learning** [[paper]](https://arxiv.org/abs/2204.04213)[[video]](https://m2d2.io/talks/m2d2/structure-aware-protein-self-supervised-learning/)
1. [KDD 2022] **GBPNet: Universal Geometric Representation Learning on Protein Structures** [[paper]](https://dl.acm.org/doi/abs/10.1145/3534678.3539441)[[code]](https://github.com/sarpaykent/GBPNet)
1. [Arxiv 2022] **Directed Weight Neural Networks for Protein Structure Representation Learning** [[paper]](https://arxiv.org/abs/2201.13299)
1. [PLOS Computational Biology 2022] **Fast protein structure comparison through effective representation learning with contrastive graph neural networks** [[paper]](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009986)[[code]](https://github.com/chunqiux/GraSR)
1. [NeurIPS 2022] **Exploring evolution-based &-free protein language models as protein function predictors** [[paper]](https://arxiv.org/pdf/2206.06583)
1. [bioRxiv 2022] **High-resolution de novo structure prediction from primary sequence** [[paper]](https://www.biorxiv.org/content/10.1101/2022.07.21.500999v1)[[code]](https://github.com/HeliXonProtein/OmegaFold)
1. [Bioinformatics  2022] **ProteinBERT: A universal deep-learning model of protein sequence and function** [[paper]](https://academic.oup.com/bioinformatics/article/38/8/2102/6502274)[[code]](https://github.com/nadavbra/protein_bert)
1. [Communications Biology 2022] **Artificial Intelligence Guided Conformational Mining of Intrinsically Disordered Proteins** [[paper]](https://www.nature.com/articles/s42003-022-03562-y)[[code]](https://github.com/aaayushg/generative_IDPs)
1. [Cell Systems 2022] **Evolutionary velocity with protein language models predicts evolutionary dynamics of diverse proteins** [[paper]](https://www.sciencedirect.com/science/article/pii/S2405471222000382)[[code]](https://github.com/brianhie/evolocity)
1. [bioRxiv  2022] **Convolutions are competitive with transformers for protein sequence pretraining** [[paper]](https://www.biorxiv.org/content/10.1101/2022.05.19.492714.abstract)[[code]](https://github.com/microsoft/protein-sequence-models)
1. [bioRxiv  2022] **Masked inverse folding with sequence transfer for protein representation learning** [[paper]](https://www.biorxiv.org/content/10.1101/2022.05.25.493516.abstract)[[code]](https://github.com/microsoft/protein-sequence-models)
1. [Nature methods 2022] **Self-supervised deep learning encodes high-resolution features of protein subcellular localization** [[paper]](https://www.nature.com/articles/s41592-022-01541-z)[[code]](https://github.com/royerlab/cytoself)
1. [bioRxiv  2022] **Language models of protein sequences at the scale of evolution enable accurate structure prediction** [[paper]](https://www.biorxiv.org/content/10.1101/2022.07.20.500902.abstract)
1. [bioRxiv  2022] **Atomic protein structure refinement using all-atom graph representations and SE(3)-equivariant graph neural networks** [[paper]](https://www.biorxiv.org/content/10.1101/2022.05.06.490934v2.abstract)
1. [Bioinformatics 2022] **Cross-Modality and Self-Supervised Protein Embedding for Compound–Protein Affinity and Contact Prediction** [[paper]](https://academic.oup.com/bioinformatics/article-abstract/38/Supplement_2/ii68/6702011)[[code]](https://github.com/Shen-Lab/CPAC)
1. [bioRxiv 2022] **COLLAPSE: A representation learning framework for identification and characterization of protein structural sites** [[paper]](https://www.biorxiv.org/content/10.1101/2022.07.20.500713v2)  
1. [bioRxiv 2022] **An Analysis of Protein Language Model Embeddings for Fold Prediction** [[paper]](https://www.biorxiv.org/content/10.1101/2022.02.07.479394v1.full)  
1. [ICLR 2022] **OntoProtein: Protein Pretraining With Gene Ontology Embedding** [[paper]](https://openreview.net/pdf?id=yfe1VMYAXa4) [[code]](https://github.com/zjunlp/OntoProtein)
1. [Bioinformatics 2022] **DistilProtBert: a distilled protein language model used to distinguish between real proteins and their randomly shuffled counterparts** [[paper]](https://academic.oup.com/bioinformatics/article/38/Supplement_2/ii95/6701995)
1. [Briefings in Bioinformatics 2022] **SPRoBERTa: protein embedding learning with local fragment modeling** [[paper]](https://academic.oup.com/bib/article-abstract/23/6/bbac401/6711410)
1. [NeurIPS 2022] **Antigen-Specific Antibody Design and Optimization with Diffusion-Based Generative Models for Protein Structures**[[paper]](https://openreview.net/forum?id=jSorGn2Tjg)[[code]](https://github.com/luost26/diffab)
1. [NeurIPS 2022] **PEER: A Comprehensive and Multi-Task Benchmark for Protein Sequence Understanding**[[paper]](https://openreview.net/forum?id=QgTZ56-zJou)[[code]](https://github.com/DeepGraphLearning/PEER_Benchmark)
1. [AAAI 2022] **Self-Supervised Pre-training for Protein Embeddings Using Tertiary Structures**[[paper]](https://ojs.aaai.org/index.php/AAAI/article/view/20636)[[code]](https://github.com/DeepGraphLearning/PEER_Benchmark)
1. [ICLR 2022] **Geometric Transformers for Protein Interface Contact Prediction**[[paper]](https://openreview.net/forum?id=CS4463zx6Hi)[[code]](https://github.com/BioinfoMachineLearning/DeepInteract)
1. [ICLR 2022] **Independent SE(3)-Equivariant Models for End-to-End Rigid Protein Docking**[[paper]](https://openreview.net/forum?id=GQjaI9mLet)[[code]](https://github.com/octavian-ganea/equidock_public)
1. [ICML 2022] **Proximal Exploration for Model-guided Protein Sequence Design**[[paper]](https://proceedings.mlr.press/v162/ren22a.html)[[code]](https://github.com/HeliXonProtein/proximal-exploration)
1. [ICML 2022] **Generating 3D Molecules for Target Protein Binding**[[paper]](https://proceedings.mlr.press/v162/liu22m.html)[[code]](https://github.com/divelab/GraphBP)
1. [ICML 2022] **Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval**[[paper]](https://proceedings.mlr.press/v162/notin22a.html)[[code]](https://github.com/OATML-Markslab/Tranception)
1. [Arxiv 2022] **DiffBP: Generative Diffusion of 3D Molecules for Target Protein Binding**[[paper]](https://arxiv.org/pdf/2211.11214.pdf)

## Year 2021

1. [Arxiv 2021] **Pre-training co-evolutionary protein representation via a pairwise masked language model** [[paper]](https://arxiv.org/abs/2110.15527)
1. [NeurIPS 2021] **Language models enable zero-shot prediction of the effects of mutations on protein function** [[paper]](https://proceedings.neurips.cc/paper/2021/hash/f51338d736f95dd42427296047067694-Abstract.html)[[code]](https://github.com/facebookresearch/esm)
1. [ICML 2021] **MSA Transformer** [[paper]](http://proceedings.mlr.press/v139/rao21a.html)[[code]](https://github.com/facebookresearch/esm)
1. [TPAMI 2021] **ProtTrans: Towards Cracking the Language of Lifes Code Through Self-Supervised Learning** [[paper]](https://ieeexplore.ieee.org/abstract/document/9477085/)[[code]](https://github.com/agemagician/ProtTrans)
1. [Arxiv 2021] **Modeling Protein Using Large-scale Pretrain Language Model** [[paper]](https://arxiv.org/abs/2108.07435)[[code]](https://github.com/THUDM/ProteinLM)
1. [IEEE Access 2021] **Pre-Training of Deep Bidirectional Protein Sequence Representations With Structural Information** [[paper]](https://ieeexplore.ieee.org/abstract/document/9529198/)[[code]](https://github.com/mswzeus/PLUS)
1. [Bioinformatics 2021] **GraphQA: protein model quality assessment using graph convolutional networks** [[paper]](https://academic.oup.com/bioinformatics/article-abstract/37/3/360/5891171)[[code]](https://github.com/baldassarreFe/protein-quality-gn)
1. [ICLR 2021] **Intrinsic-extrinsic convolution and pooling for learning on 3d protein structures** [[paper]](https://arxiv.org/abs/2007.06252)[[code]](https://github.com/phermosilla/IEConv_proteins)
1. [ICLR 2021] **Learning from Protein Structure with Geometric Vector Perceptrons** [[paper]](https://arxiv.org/abs/2009.01411)[[code]](https://github.com/drorlab/gvp)
1. [NeurIPS 2021] **Multi-Scale Representation Learning on Proteins** [[paper]](https://proceedings.neurips.cc/paper/2021/hash/d494020ff8ec181ef98ed97ac3f25453-Abstract.html)[[code]](https://github.com/vsomnath/holoprot)
1. [PNAS 2021] **Neural networks to learn protein sequence–function relationships from deep mutational scanning data** [[paper]](https://www.pnas.org/doi/abs/10.1073/pnas.2104878118)[[code]](https://github.com/gitter-lab/nn4dms)
1. [bioRxiv 2021] **LM-GVP: A Generalizable Deep Learning Framework for Protein Property Prediction from Sequence and Structure** [[paper]](https://www.biorxiv.org/content/10.1101/2021.09.21.460852v1.abstract)[[code]](https://github.com/aws-samples/lm-gvp)
1. [Cell Systems 2021] **Learning the protein language: Evolution, structure, and function** [[paper]](https://www.sciencedirect.com/science/article/pii/S2405471221002039)[[code]](https://github.com/tbepler/prose)
1. [Nature Communications 2021] **Structure-based protein function prediction using graph convolutional networks** [[paper]](https://www.nature.com/articles/s41467-021-23303-9)[[code]](https://github.com/flatironinstitute/DeepFRI)
1. [KDD 2021] **Geometric Graph Representation Learning on Protein Structure Prediction** [[paper]](https://dl.acm.org/doi/10.1145/3447548.3467323)
1. [Arxiv 2021] **Adversarial Contrastive Pre-training for Protein Sequences** [[paper]](https://arxiv.org/pdf/2102.00466)
1. [Emerg Top Life Sci 2021] **Graph representation learning for structural proteomics** [[paper]](https://portlandpress.com/emergtoplifesci/article/5/6/789/229961/Graph-representation-learning-for-structural)
1. [Arxiv 2021] **Graph Representation Learning in Biomedicine** [[paper]](https://arxiv.org/abs/2104.04883)
1. [Applied Sciences 2021] **GraphMS:Drug Target Prediction Using Graph Representation Learning with Substructures** [[paper]](https://www.mdpi.com/2076-3417/11/7/3239)[[code]](https://github.com/FangpingWan/NeoDTI)
1. [JMC 2021] **InteractionGraphNet: A Novel and Efficient Deep Graph Representation Learning Framework for Accurate Protein−Ligand Interaction Predictions** [[paper]](https://pubs.acs.org/doi/pdf/10.1021/acs.jmedchem.1c01830)[[code]](https://github.com/zjujdj/InteractionGraphNet/tree/master)
1. [bioRxiv 2021] **Self-Supervised Representation Learning of Protein Tertiary Structures (PtsRep) and Its Implications for Protein Engineering** [[paper]](https://www.biorxiv.org/content/10.1101/2020.12.22.423916.abstract)
1. [Algorithms 2021] **Capturing Protein Domain Structure and Function Using Self-Supervision on Domain Architectures** [[paper]](https://www.mdpi.com/1999-4893/14/1/28)[[code]](https://github.com/damianosmel/dom2vec)
1. [bioRxiv 2021] **Combining evolutionary and assay-labelled data for protein fitness prediction** [[paper]](https://www.biorxiv.org/content/10.1101/2021.03.28.437402.abstract)
1. [Science 2021] **Accurate prediction of protein structures and interactions using a three-track neural network** [[paper]](https://www.ipd.uw.edu/wp-content/uploads/2021/07/Baek_etal_Science2021_RoseTTAFold.pdf)[[code]](https://github.com/RosettaCommons/RoseTTAFold)
1. [Nature 2021] **Highly accurate protein structure prediction with AlphaFold** [[paper]](https://www.nature.com/articles/s41586-021-03819-2)[[code]](https://github.com/deepmind/alphafold)
1. [IEEE TCBB 2021] **Sequence representations and their utility for predicting protein-protein interactions** [[paper]](https://ieeexplore.ieee.org/abstract/document/9661301/)
1. [Bioinformatics 2021] **Unsupervised protein embeddings outperform hand-crafted sequence and structure features at predicting molecular function** [[paper]](https://academic.oup.com/bioinformatics/article-abstract/37/2/162/5892762)[[code]](https://github.com/stamakro/GCN-for-Structure-and-Function)
1. [CVPR 2021] **Fast end-to-end learning on protein surfaces** [[paper]](https://openaccess.thecvf.com/content/CVPR2021/html/Sverrisson_Fast_End-to-End_Learning_on_Protein_Surfaces_CVPR_2021_paper.html)
1. [Briefings in Functional Genomics 2021] **Pretraining model for biological sequence data** [[paper]](https://academic.oup.com/bfg/article-abstract/20/3/181/6287893)
1. [bioRxiv 2021] **Toward More General Embeddings for Protein Design: Harnessing Joint Representations of Sequence and Structure** [[paper]](https://www.biorxiv.org/content/10.1101/2021.09.01.458592.abstract)
1. [NeurIPS 2021] **Neural Distance Embeddings for Biological Sequences** [[paper]](https://proceedings.neurips.cc/paper/2021/hash/9a1de01f893e0d2551ecbb7ce4dc963e-Abstract.html)[[code]](https://github.com/gcorso/NeuroSEED)
1. [Computational Biology and Chemistry 2021] **Convolutional neural networks with image representation of amino acid sequences for protein function prediction** [[paper]](https://www.sciencedirect.com/science/article/abs/pii/S147692712100061X)[[code]](https://github.com/swakkhar/ProtConv)
1. [bioRxiv  2021] **Distillation of MSA Embeddings to Folded Protein Structures with Graph Transformers** [[paper]](https://www.biorxiv.org/content/10.1101/2021.06.02.446809.abstract)  
1. [chemRxiv 2021] **Identification of Enzymatic Active Sites with Unsupervised Language Modeling** [[paper]](https://chemrxiv.org/engage/chemrxiv/article-details/61d7f0506be4200bbf24829e)  
1. [bioRxiv 2021] **Deciphering the language of antibodies using self-supervised learning** [[paper]](https://www.biorxiv.org/content/10.1101/2021.11.10.468064v1)  
1. [bioRxiv 2021] **Hydrogen bonds meet self-attention: all you need for general-purpose protein structure embedding** [[paper]](https://www.biorxiv.org/content/10.1101/2021.01.31.428935v2)   
1. [bioRxiv 2021] **Improving Generalizability of Protein Sequence Models with Data Augmentations** [[paper]](https://www.biorxiv.org/content/10.1101/2021.02.18.431877v1)  



## Year 2020

1. [BCB 2020] **Transforming the Language of Life: Transformer Neural Networks for Protein Prediction Tasks** [[paper]](https://dl.acm.org/doi/abs/10.1145/3388440.3412467)[[code]](https://github.com/annambiar/PRoBERTa)
2. [Bioinformatics 2020] **UDSMProt: universal deep sequence models for protein classification** [[paper]](https://academic.oup.com/bioinformatics/article/36/8/2401/5698270)[[code]](https://github.com/nstrodt/UDSMProt)
3. [bioRxiv 2020] **Self-Supervised Contrastive Learning of Protein Representations By Mutual Information Maximization** [[paper]](https://www.biorxiv.org/content/10.1101/2020.09.04.283929v1)
4. [bioRxiv 2020] **End-to-end multitask learning, from protein language to protein features without alignments** [[paper]](https://www.biorxiv.org/content/10.1101/864405.abstract)
5. [bioRxiv 2020] **Language modelling for biological sequences – curated datasets and baselines** [[paper]](https://www.biorxiv.org/content/10.1101/2020.03.09.983585v1.abstract)[[code]](https://github.com/alrojo/UniLanguage)
6. [NeurIPS 2020] **Is Transfer Learning Necessary for Protein Landscape Prediction?** [[paper]](https://arxiv.org/abs/2011.03443)
7. [PNAS 2020] **Biological structure and function emerge from scaling unsupervised learning to 250 million protein sequences** [[paper]](https://www.pnas.org/doi/pdf/10.1073/pnas.2016239118)[[code]](https://github.com/facebookresearch/esm)
8. [Arxiv 2020] **Profile Prediction: An Alignment-Based Pre-Training Task for Protein Sequence Models** [[paper]](https://arxiv.org/abs/2012.00195)
9. [Arxiv 2020] **ProGen: Language Modeling for Protein Generation** [[paper]](https://arxiv.org/abs/2004.03497)[[code]](https://github.com/lucidrains/progen)
10. [bioRxiv 2020] **Evaluation of Methods for Protein Representation Learning: A Quantitative Analysis** [[paper]](https://www.biorxiv.org/content/10.1101/2020.10.28.359828.abstract)
11. [NAR Genomics and Bioinformatics 2020] **Mutation effect estimation on protein–protein interactions using deep contextualized representation learning** [[paper]](https://academic.oup.com/nargab/article-abstract/2/2/lqaa015/5781175)[[code]](https://github.com/guangyu-zhou/MuPIPR)
12. [CSBJ 2020] **Representation learning applications in biological sequence analysis** [[paper]](https://www.sciencedirect.com/science/article/pii/S2001037021002208)
13. [bioRxiv 2020] **TripletProt: Deep Representation Learning of Proteins based on Siamese Networks** [[paper]](https://www.biorxiv.org/content/10.1101/2020.05.11.088237.abstract)
14. [RCMB 2020] **Evolutionary context-integrated deep sequence modeling for protein engineering** [[paper]](https://link.springer.com/chapter/10.1007/978-3-030-45257-5_30)
15. [Arxiv 2020] **What is a meaningful representation of protein sequences?** [[paper]](https://arxiv.org/abs/2012.02679)[[code]](https://github.com/MachineLearningLifeScience/meaningful-protein-representations)
16. [bioRxiv 2020] **Transformer protein language models are unsupervised structure learners** [[paper]](https://www.biorxiv.org/content/10.1101/2020.12.15.422761v1.abstract)[[code]](https://github.com/facebookresearch/esm)
17. [bioRxiv 2020] **Deep learning-based prediction of protein structure using learned representations of multiple sequence alignments** [[paper]](https://www.researchgate.net/profile/Andy-Lau-9/publication/346595356_Deep_learning-based_prediction_of_protein_structure_using_learned_representations_of_multiple_sequence_alignments/links/5fc8dd1fa6fdcc697bd860ba/Deep-learning-based-prediction-of-protein-structure-using-learned-representations-of-multiple-sequence-alignments.pdf)



## Year before-2019

1. [Cell 2019] **A High Efficient Biological Language Model for Predicting Protein–Protein Interactions** [[paper]](https://www.mdpi.com/406658)[[code]](https://figshare.com/s/b35a2d3bf442a6f15b6e)
1. [Nature Method 2019] **Unified rational protein engineering with sequence-only deep representation learning** [[paper]](https://www.nature.com/articles/s41592-019-0598-1)[[code]](https://github.com/churchlab/UniRep)
1. [NeurIPS 2019] **Evaluating Protein Transfer Learning with TAPE** [[paper]](https://arxiv.org/abs/1906.08230)[[code]](https://github.com/songlab-cal/tape)
1. [Nature communications 2019] **Deciphering protein evolution and fitness landscapes with latent space models** [[paper]](https://www.nature.com/articles/s41467-019-13633-0?ref=https://githubhelp.com)[[code]](https://github.com/xqding/PEVAE_Paper)
1. [bioRxiv 2019] **DeepPrime2Sec: Deep Learning for Protein Secondary Structure Prediction from the Primary Sequences** [[paper]](https://www.biorxiv.org/content/10.1101/705426v1)[[code]](https://github.com/ehsanasgari/DeepPrime2Sec)
1. [ACS Nano 2019] **A Self-Consistent Sonification Method to Translate Amino Acid Sequences into Musical Compositions and Application in Protein Design Using Artificial Intelligence** [[paper]](https://pubs.acs.org/doi/10.1021/acsnano.9b02180)  
2. [bioRxiv 2019] **Augmenting protein network embeddings with sequence information** [[paper]](https://www.biorxiv.org/content/10.1101/730481v3)
3. [Nature Machine Intelligence 2019] **Leveraging implicit knowledge in neural networks for functional dissection and engineering of proteins.** [[paper]](https://www.nature.com/articles/s42256-019-0049-9)[[code]](https://github.com/juzb/DeeProtein)
4. [bioRxiv 2019] **Modeling the Language of Life – Deep Learning Protein Sequences** [[paper]](https://www.biorxiv.org/content/10.1101/614313v2)[[code]](https://github.com/Rostlab/SeqVec)
5. [ICLR 2019] **Learning protein sequence embeddings using information from structure** [[paper]](https://arxiv.org/abs/1902.08661)[[code]](https://github.com/tbepler/protein-sequence-embedding-iclr2019)
6. [BIBM 2019] **GraphCPI: Graph Neural Representation Learning for Compound-Protein Interaction** [[paper]](https://ieeexplore.ieee.org/abstract/document/8983267/)
7. [Bioinformatics 2018] **Learned protein embeddings for machine learning** [[paper]](https://academic.oup.com/bioinformatics/article/34/15/2642/4951834)[[code]](https://github.com/fhalab/embeddings_reproduction/)
8. [Bioinformatics 2018] **Deep convolutional networks for quality assessment of protein folds** [[paper]](https://academic.oup.com/bioinformatics/article/34/23/4046/5040325)[[code]](https://github.com/lamoureux-lab/3DCNN_MQA)
9. [bioRxiv 2018] **Deep Semantic Protein Representation for Annotation, Discovery, and Engineering** [[paper]](https://www.biorxiv.org/content/10.1101/365965.abstract)[[code]](https://github.com/syntheticgenomics/sgidspace)
10. [bioRxiv 2017] **Predicting Protein Binding Affinity With Word Embeddings and Recurrent Neural Networks** [[paper]](https://www.biorxiv.org/node/37703.abstract)[[code]](https://github.com/carlomazzaferro/mhcPreds)
11. [Arxiv 2017] **Variational auto-encoding of protein sequences** [[paper]](https://arxiv.org/abs/1712.03346)[[code]](https://github.com/samsinai/VAE_protein_function)
12. [Arxiv 2016] **Distributed Representations for Biological Sequence Analysis** [[paper]](https://arxiv.org/abs/1608.05949)
13. [Bioinformatics 2015] **ProFET: Feature engineering captures high-level protein functions** [[paper]](https://academic.oup.com/bioinformatics/article/31/21/3429/194375?login=false)  



## Related Awesome

* [awesome-graph-representation-learning](https://github.com/zlpure/awesome-graph-representation-learning)
* [awesome-graph-self-supervised-learning](https://github.com/LirongWu/awesome-graph-self-supervised-learning)
* [awesome-self-supervised-gnn](https://github.com/ChandlerBang/awesome-self-supervised-gnn)
* [awesome-self-supervised-learning-for-graphs](https://github.com/SXKDZ/awesome-self-supervised-learning-for-graphs)
* [awesome-AI-based-protein-design](https://github.com/opendilab/awesome-AI-based-protein-design)



## Citation

If you find this project useful for your research, please use the following BibTeX entry.

```
@article{wu2022survey,
  title={A Survey on Protein Representation Learning: Retrospect and Prospect},
  author={Wu, Lirong and Huang, Yufei and Lin, Haitao and Li, Stan Z},
  journal={arXiv preprint arXiv:2301.00813},
  year={2022}
}
```



## Contact

If you have any issue about this work, please feel free to contact me by email:

- Lirong Wu: wulirong@westlake.edu.cn
- Yufei Huang: huangyufei@westlake.edu.cn
- Bozhen Hu: hubozhen@westlake.edu.cn
